client/signup/controller.js

```diff
@@ -225,6 +226,19 @@ export default {
 
 		store.set( 'signup-locale', localeFromParams );
 
+		const isOnboardingFlow = flowName === 'onboarding';
+		if ( isOnboardingFlow ) {
+			const stepperOnboardingExperimentAssignment = await loadExperimentAssignment(
+				'calypso_signup_onboarding_flow_stepper'
+			);
+			if ( stepperOnboardingExperimentAssignment.variationName === 'stepper' ) {
+				const lang = localeFromParams ?? localeFromStore;
+				window.location.replace(
+					window.location.origin + `/setup/onboarding/${ lang }` + window.location.search
```

Comment:
1. Do we need the `lang` suffix?
2. Do we lose anything from pathname with this? e.g. a request for `/start/plans` should redirect to `/setup/onboarding/plans` right?

Reply:
> Do we need the lang suffix?

I believe so, as right now it is used to determine locale. It might need an improvement, as to determine locale as it happens in start. But for now this is the way I found. cc @daledupreez 

> Do we lose anything from pathname with this

You're right. I pushed a new version that should be more robust. I didn't over-engineered as it's only temporary. I could have used the `getStepUrl` function forcing the framework. I preferred not to